--- config.h	2011-09-13 00:33:28.062630721 +0300
+++ ../config.h	2011-09-13 00:39:52.652623181 +0300
@@ -138,8 +138,6 @@
 #define HAVE_DLOPEN 1
 #define HAVE_SIGPROCMASK 1
 #define HAVE_SIGACTION 1
-#define HAVE__SETJMP 1
-#define HAVE__LONGJMP 1
 #define HAVE_SETSID 1
 #define HAVE_TELLDIR 1
 #define HAVE_SEEKDIR 1
@@ -155,8 +153,6 @@
 #define HAVE_SETGID 1
 #define HAVE_SETENV 1
 #define HAVE_UNSETENV 1
-#define RUBY_SETJMP(env) _setjmp(env)
-#define RUBY_LONGJMP(env,val) _longjmp(env,val)
 #define RUBY_JMP_BUF jmp_buf
 #define HAVE_STRUCT_TM_TM_ZONE 1
 #define HAVE_TM_ZONE 1
@@ -187,3 +183,13 @@
 #define RUBY_ARCHLIB "/usr/local/lib/ruby/1.8/x86_64-linux"
 #define RUBY_SITE_ARCHLIB "/usr/local/lib/ruby/site_ruby/1.8/x86_64-linux"
 #define RUBY_VENDOR_ARCHLIB "/usr/local/lib/ruby/vendor_ruby/1.8/x86_64-linux"
+
+/* emscripten */
+#define RUBY_SETJMP(env) emscripten_setjmp(env)
+#define RUBY_LONGJMP(env,val) emscripten_longjmp(env,val)
+#include <setjmp.h>
+int emscripten_setjmp(jmp_buf env);
+void emscripten_longjmp(jmp_buf env, int ret);
+#define ANYARGS ...
+#include <unistd.h>
+/* /emscripten */
